### PR TITLE
drivers: sensor: lsm303dlhc_mag: Fix gain factor

### PR DESCRIPTION
**Pull request by microbuilder**
The magnetometer on the LSM303DLHC has a different gain conversion
factor for LSB to Gauss for the Z axis than it does for X, Y. This
commit takes into account the different conversion factors, and
adds the correct coefficients for each gain setting and axis.

Signed-off-by: Kevin Townsend <kevin.townsend@linaro.org>

### CODE REVIEW

**Comment by microbuilder** [comment]
See table 3 in the datasheet for details:

Screenshot 2021-09-28 at 13 14 01

**Review by avisconti — APPROVED** [review]
Very clean indeed... Thanks!

File: drivers/sensor/lsm303dlhc_magn/lsm303dlhc_magn.h [hunk 1]
```diff
@@ -37,19 +37,26 @@
 					LSM303DLHC_MAGN_ODR_SHIFT)
 
 #if	(CONFIG_LSM303DLHC_MAGN_RANGE == 1)
-	#define LSM303DLHC_MAGN_LSB_GAUSS	1100
+	#define LSM303DLHC_MAGN_LSB_GAUSS_XY	1100
+	#define LSM303DLHC_MAGN_LSB_GAUSS_Z		980
```
Comment (erwango) [inline drivers/sensor/lsm303dlhc_magn/lsm303dlhc_magn.h:41]: Nit: indentation

**Review by erwango — APPROVED** [review]

File: drivers/sensor/lsm303dlhc_magn/lsm303dlhc_magn.h (hunk 1 above)
Comment (cfriedt) [reply drivers/sensor/lsm303dlhc_magn/lsm303dlhc_magn.h:41]: @microbuilder - can you fix this one last nit?

Comment (erwango) [reply drivers/sensor/lsm303dlhc_magn/lsm303dlhc_magn.h:41]: @cfriedt This is not blocking, I approved the change anyway. This is left to @microbuilder appreciation.

Comment (microbuilder) [reply drivers/sensor/lsm303dlhc_magn/lsm303dlhc_magn.h]: Pushed a fix out.

**Pull request merged**